File: ws/src/main/scala/de/thm/ii/fbs/model/CheckResult.scala
package de.thm.ii.fbs.model

/**
  * Check result of a single check
  * @param exitCode Exit code of the check
  * @param resultText Output text of the checker
  * @param checkerType The cheker type that checked the submission
  * @param configurationId Checker configuration id
  */
case class CheckResult(exitCode: Int = 1, resultText: String, checkerType: String, configurationId: Int)

File: modules/fbs-core/api/src/main/kotlin/de/thm/ii/fbs/model/v2/CheckResult.kt
package de.thm.ii.fbs.model.v2

import com.fasterxml.jackson.databind.JsonNode

/**
 * Check result of a single check
 *
 * @param exitCode        Exit code of the check
 * @param resultText      Output text of the checker
 * @param checkerType     The checker type that checked the submission
 * @param configurationId Checker configuration id
 * @param extInfo         Checker extended information
 */
data class CheckResult(val exitCode: Int = 1, val resultText: String, val checkerType: String, val configurationId: Int, val extInfo: JsonNode)
